File: CodingPractice-master/leetcode/algorithms/cpp/0231_PowerOfTwo.cpp
/*
* @Author: Yan_Daojiang
* @Date:   2019-05-17 10:30:00
* @Last Modified by:   Yan_Daojiang
* @Last Modified time: 2019-05-17 10:30:15
*/
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if(n==0)
            return false;
        else{
            while(n!=0){
                if(n%2!=0&&n!=1)
                    return false;
                if(n==1)
                    break;
                 n/=2;
            }
        }
        return true;
    }
};

File: CodingPractice-master/leetcode/algorithms/python/0231_PowerOfTwo.py
# -*- coding: utf-8 -*-
# @Author: Yan_Daojiang
# @Date:   2019-05-17 10:33:38
# @Last Modified by:   Yan_Daojiang
# @Last Modified time: 2019-05-17 10:33:46
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n==0:
            return False
        else:
            while n!=0:
                if n%2!=0 and n!=1:
                    return False
                if n==1:
                    return True
                n=int(n/2)
        return True